### PR TITLE
Fix tests and reanimate jenkins

### PR DESCRIPTION
**Pull request by mussbach**
This PR collects a bunch of changes to fix the unit tests and deprecations. Moreover this PR is used to rebuild the connection to CI

### How to review/test
unit Test should not fail. There are still some errors, but no failures should appear

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/TransformMessageBagService.php [hunk 1]
```diff
@@ -32,11 +33,11 @@ class TransformMessageBagService
 
     public function __construct(
         MessageBagInterface $messageBag,
-        FlashBagInterface $flashBag,
+        RequestStack $requestStack,
         RouterInterface $router
     ) {
         $this->messageBag = $messageBag;
-        $this->flashBag = $flashBag;
+        $this->flashBag = $requestStack->getCurrentRequest()?->getSession()->getFlashBag();
```
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/TransformMessageBagService.php]: [c717127](https://github.com/demos-europe/demosplan-core/pull/1465/commits/c717127a75cc6e9256bd8424be4be99c577b0347)

File: demosplan/DemosPlanCoreBundle/Logic/User/UserMapperDataportGateway.php [hunk 2]
```diff
@@ -343,7 +343,7 @@ public function decode($string)
         $string = Utf8::filter($string);
         $string = html_entity_decode($string, ENT_QUOTES);
 
-        return trim($string);
+        return trim($string ?? '');
```
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/User/UserMapperDataportGateway.php]: aa2bfc98

File: demosplan/DemosPlanCoreBundle/Logic/TransformMessageBagService.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/TransformMessageBagService.php]: I would argue that this is a problem in the framework. The fetch of the session could be simplified though 5b08789e

[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/TransformMessageBagService.php]: see above

**Review by martendemos — APPROVED** [review]
code review only

**Pull request merged**